lib/migration.js

```diff
@@ -461,8 +461,8 @@ function mixinMigration(MySQL, mysql) {
 
   function columnType(p, defaultType) {
     var dt = defaultType;
-    if (p.dataType) {
-      dt = String(p.dataType);
```

Comment:
for backwards compatible,  I pick a case here for discussion:

```
"modelName": {
  "type" : "someType",
  "dataType": "someDataType",
  "mysql" : {
   "dataType": "someOtherType"
  }
}
```
what column name should be used in the database ? IMHO, it should be "someOtherType", because it is defined in mysql and it is expected to be.  We should introduce new test(s) or change existing test(s) to make it clear.